### PR TITLE
[pipeline] parallelize cross join operators

### LINKED ISSUE #1174: [pipeline] parallelize Cross Join Operator
## Enhancement
Nowadays, degree_of_parallelism (DOP) of `CrossJoinRightSinkOperator` and `CrossJoinLeftOperator` can only be one.
`CrossJoinRightSinkOperator` collects all the rows to a chunk (BuildRows), and `CrossJoinLeftOperator` uses `_build_rows_index` and `_probe_rows_index` to traverse BuildRows and ProbeRows to get output chunk.
CrossJoin-Probe

Each `CrossJoinRightSinkOperator` can maintain its own BuildRows instead of collecting them to one BuildRows. And `CrossJoinLeftOperator` uses an additional index `_probe_which_build_index` to maintain which BuildRows is traversed.
CrossJoin-Probe2

### PR DESCRIPTION
**Pull request by ZiheLiu**
Close #1174.

### CODE REVIEW

**Comment by ZiheLiu** [comment]
run starrocks_clang-forma

**Comment by ZiheLiu** [comment]
run starrocks_clang-format

**Review by liuyehcf — DISMISSED** [review]

File: be/src/exec/vectorized/cross_join_node.cpp [hunk 1]
```diff
@@ -537,33 +537,31 @@ void CrossJoinNode::_init_chunk(ChunkPtr* chunk) {
...
-
     // step 0: construct pipeline end with cross join right operator.
-    OpFactories operator_before_cross_join_right = _children[1]->decompose_to_pipeline(context);
-    operator_before_cross_join_right =
-            context->maybe_interpolate_local_passthrough_exchange(operator_before_cross_join_right);
+    OpFactories right_ops = _children[1]->decompose_to_pipeline(context);
 
     // communication with CrossJoinLeft through shared_datas.
+    auto* right_source = down_cast<SourceOperatorFactory*>(right_ops[0].get());
+    std::shared_ptr<pipeline::CrossJoinContext> cross_join_context =
```
Comment (Pslydhh) [inline be/src/exec/vectorized/cross_join_node.cpp:545]: Maybe It's better to use **auto cross_join_context**, It's better to keep consistent with others.

**Comment by ZiheLiu** [comment]
run starrocks_clang-format

**Comment by ZiheLiu** [comment]
run starrocks_be_unittest

**Comment by ZiheLiu** [comment]
run starrocks_be_unittest

**Review by kangkaisen — APPROVED** [review]

**Review by Pslydhh — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Pull request merged**